src/Sylius/Bundle/ApiBundle/DependencyInjection/SyliusApiExtension.php

```diff
@@ -31,10 +31,12 @@ public function load(array $configs, ContainerBuilder $container): void
 
         $loader->load('services.xml');
 
-        if (!$container->hasParameter('enable_swagger') || !$container->getParameter('enable_swagger')) {
-            return;
+        if ($container->hasParameter('enable_swagger') && $container->getParameter('enable_swagger')) {
+            $loader->load('integrations/swagger.xml');
         }
 
-        $loader->load('integrations/swagger.xml');
+        if (class_exists(\ApiPlatform\Core\OpenApi\OpenApi::class)) {
```

Comment:
I'm testing for class existence here because the `api_platform.openapi.factory` service apparently is not defined yet and `$container->has('api_platform.openapi.factory')` returns `false`.

To avoid service not found errors when the service indeed doesn't exist for some reason, I've added `decoration-on-invalid="ignore"` to all service definitions. 

Not sure if this is the best approach though.